src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,702 @@
+  @Prop({ reflect: true }) hourDisplayFormat: "12" | "24" = "12";
+
+  /** The minute value */
+  @Prop({ reflect: true, mutable: true }) minute?: string = "--";
+
+  /** The second value */
+  @Prop({ reflect: true, mutable: true }) second?: string = "--";
+
+  /** The scale (size) of the time picker */
+  @Prop({ reflect: true }) scale: Scale = "m";
+
+  /** number that specifies the granularity that the value must adhere to */
+  @Prop({ reflect: true }) step = 60;
+
+  /** The color theme of the time-picker */
+  @Prop({ reflect: true }) theme: Theme = "light";
+
+  @Watch("hour")
+  hourChanged(newHour: string): void {
+    if (this.hourDisplayFormat === "12" && newHour !== "--") {
+      this.ampm = this.getAmPm();
+    }
+  }
+
+  @Watch("hour")
+  @Watch("minute")
+  @Watch("second")
+  timeChangeHandler(newValue: string, oldValue: string): void {
+    if (this.timeChanged && newValue !== oldValue) {
+      this.calciteTimePickerChange.emit(this.getTime());
+      this.timeChanged = false;
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  private activeEl: HTMLSpanElement;
+
+  private amPmEl: HTMLSpanElement;
+
+  private hourEl: HTMLSpanElement;
+
+  private minuteEl: HTMLSpanElement;
+
+  private secondEl: HTMLSpanElement;
+
+  private timeChanged = false;
+
+  // --------------------------------------------------------------------------
+  //
+  //  State
+  //
+  // --------------------------------------------------------------------------
+
+  /** The am/pm value */
+  @State() ampm: AmPm = "--";
+
+  /** The display version of the hour */
+  @State() displayHour: string = this.getDisplayHour();
+
+  /** Whether the hour is being edited while focused */
+  @State() editingHourWhileFocused = false;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * @internal
+   */
+  @Event() calciteTimePickerBlur: EventEmitter<Time>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteTimePickerChange: EventEmitter<Time>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteTimePickerFocus: EventEmitter<Time>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("blur")
+  hostBlurHandler(): void {
+    this.calciteTimePickerBlur.emit();
+  }
+
+  @Listen("focus")
+  hostFocusHandler(): void {
+    this.calciteTimePickerFocus.emit();
+  }
+
+  @Listen("keydown")
+  keyDownHandler(event: KeyboardEvent): void {
+    switch (this.activeEl) {
+      case this.hourEl:
+        if (event.key === "ArrowRight") {
+          this.minuteEl.focus();
```

Comment:
maybe need to add these as options on setFocus()

Reply:
Done: https://github.com/Esri/calcite-components/pull/1736/commits/0e76fc7333de77af7a5ac63d8bf5ba7841baa235